File: samples/sample-app-storage/src/main/java/com/azure/android/storage/sample/ContainerBlobsPaginationViewModel.java
package com.azure.android.storage.sample;

import com.azure.android.storage.blob.StorageBlobAsyncClient;
import com.azure.android.storage.blob.models.BlobItem;
import com.azure.android.storage.sample.core.util.paging.PaginationDescriptionRepository;
import com.azure.android.storage.sample.core.util.paging.PaginationOptions;
import com.azure.android.storage.sample.core.util.paging.PaginationViewModel;

import java.util.concurrent.Executors;

public class ContainerBlobsPaginationViewModel
        extends PaginationViewModel<PaginationDescriptionRepository<BlobItem, String>, BlobItem, String> {

    public ContainerBlobsPaginationViewModel(StorageBlobAsyncClient storageBlobAsyncClient) {
        super(getRepository(storageBlobAsyncClient));
    }

    private static PaginationDescriptionRepository<BlobItem, String> getRepository(StorageBlobAsyncClient storageBlobAsyncClient) {
        PaginationOptions paginationOptions = new PaginationOptions(Executors.newFixedThreadPool(3))
               .enableInteractiveLogin(true);
        return new ContainerBlobsPaginationRepository(storageBlobAsyncClient, paginationOptions);
    }
}

File: samples/sample-app-storage-kotlin/src/main/java/com/azure/android/storage/sample/kotlin/ContainerBlobsPaginationViewModel.kt
package com.azure.android.storage.sample.kotlin

import com.azure.android.storage.blob.StorageBlobAsyncClient
import com.azure.android.storage.blob.models.BlobItem
import java.util.concurrent.Executors
import com.azure.android.storage.sample.kotlin.core.util.paging.PaginationViewModel;
import com.azure.android.storage.sample.kotlin.core.util.paging.PaginationDescriptionRepository;
import com.azure.android.storage.sample.kotlin.core.util.paging.PaginationOptions;

class ContainerBlobsPaginationViewModel(storageBlobAsyncClient: StorageBlobAsyncClient)
    : PaginationViewModel<PaginationDescriptionRepository<BlobItem, String>, BlobItem, String>(getRepository(storageBlobAsyncClient)) {

    companion object {
        private fun getRepository(storageBlobAsyncClient: StorageBlobAsyncClient): PaginationDescriptionRepository<BlobItem, String> {
            val paginationOptions: PaginationOptions = PaginationOptions(Executors.newFixedThreadPool(3))
                .enableInteractiveLogin(true)
            return ContainerBlobsPaginationRepository(storageBlobAsyncClient, paginationOptions)
        }
    }
}
